stylix/mk-target.nix

```diff
@@ -0,0 +1,102 @@
+{
+  name,
+  humanName,
+  autoEnable ? true,
+  extraOptions ? { },
+  configElements ? [ ],
+  generalConfig ? null,
+}:
+{ config, lib, ... }:
+let
+  cfg = config.stylix.targets.${name};
+in
+{
+  options.stylix.targets.${name} = lib.attrsets.unionOfDisjoint {
+    enable = config.lib.stylix.mkEnableTarget humanName autoEnable;
+  } extraOptions;
+
+  config =
+    let
+      args =
+        fn:
+        let
+          args = lib.functionArgs fn;
+          invalidArgs = lib.filterAttrs (_: value: value) args;
+          invalidArgsString = builtins.concatStringsSep ", " (
+            builtins.attrNames invalidArgs
+          );
+        in
+        lib.throwIf (invalidArgs != { })
+          "function arguments should not have default values: ${invalidArgsString}"
+          (builtins.attrNames args);
+      provideStylixArgs =
+        args':
+        (builtins.listToAttrs (
+          map (
+            arg:
+            if arg == "cfg" then
+              {
+                name = "cfg";
+                value = cfg;
+              }
+            else if arg == "colors" then
+              {
+                name = arg;
+                value = config.lib.stylix.${arg};
+              }
+            else
+              {
+                name = arg;
+                value = config.stylix.${arg};
+              }
+          ) args'
+        ));
```

Comment:
This looks very similar to some of the other definitions below, maybe identical?

I believe this could be factored out into some private helper functions defined in a `let in` block somewhere.

Also, nested calls to multiple functions are _sometimes_ neater with `lib.pipe`; worth playing with, but not critical.